src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -2543,162 +2588,165 @@ private static Type GetDataType(EventMetadata eventData, int parameterId)
         /// </summary>
         internal void DoCommand(EventCommandEventArgs commandArgs)
         {
-            // PRECONDITION: We should be holding the EventListener.EventListenersLock
-            // We defer commands until we are completely inited.  This allows error messages to be sent.
-            Debug.Assert(m_completelyInited);
+            if (IsSupported)
```

Comment:
I think it'd be better to have code like

```
if (!IsSupported)
	return;
```

instead of these massive indentations